week-14/practices/05-Navigation/README.md

```diff
@@ -1,22 +1,153 @@
+
+Now add list-items and NavLinks for the other two routes you created.
+
+Take a look in the Browser. Notice that the active link is now red? In your
+`index.css` you have an `.active` defined. NavLink automatically sees this and
+applies it to the active link. Click each item and notice how it changes.
+
+## Phase 4: ActiveClassName and ActiveStyle
+
+Now you are going to change the class that is called on the active link using
+the `activeClassName` attribute. Take a look in the `index.css` and notice that
+there is a purple class. Now, in your App.js, add `activeClassName='purple'` to
+each NavLink. In the browser, your active link should now, also, have a border
+around it.
+
+Next, inside your NavLinks you are going to use the `activeStyle` attribute. It
+will work in the way that inline styling works for css. Add `activeStyle={{ fontWeight: 'bold' }}` to each NavLink. When you go to the browser, you should
+notice that each active link now has both a border and a bold font.
+
+## Phase 5: Redirect
+
+Sometimes you only want to allow access to a component when a user is logged in.
+Otherwise, you want to navigate the user in another direction. Using the
+Redirect component can help you do this.
+
+In your App.js just above the Page Not Found route. Add a router that looks like
+this:
+
+```js
+<Route path='/not-logged-in'>
+  <h1> You Must Be Logged In to Enter.</h1>
+</Route>
+```
+
+Next, in your Stocks.js file, create a variable called loggedIn and assign it to
+false.
+
+Add an `if` statement that redirects to the `/not-logged-in` url when the
+`loggedIn` variable is false.
+
+Test it in your Browser by clicking on the Stocks link. It should redirect you
+to the Not Logged in route.
+
+Now, in Stocks.js, set the `loggedIn` variable to true and click on the Stocks
+Link. You should be able to view the Stocks component.
+
+## History API & useHistory Hook for Navigation
+
+You will use the useHistory hook to navigate after some transaction has taken
+place. In your Stocks component, you want to navigate back to the Home Page
+after you click a button.
+
+First, under the `h1` tag in your JSX, create a button with an `onClick` event
+listener that looks like this:
+
+```js
+<button onClick={handleClick}>Home</button>`
+```
+
+Event listeners differ in React in that the event listener is camel cased and
+will always be assigned to a function.
+
+Here you have set the `onClick` to take the function handleClick but you have
+not written it yet.
+
+Above your return, create a function called `handleClick`.
+It should alert the user with a message and then navigate home.
+
+```js
+const handleClick = () => {
+  window.alert('Sending info to the DB!');
+};
+```
+
+Test in the browser.
+
+Now, you want to redirect after you click the ok button in the alert box.
+
+import the `useHistory` hook from `react-router-dom`
+
+In your component, evoke the useHistory hook and assign it to a new variable
```

Comment:
**Line 136:** Maybe "invoke" instead of "evoke"

Reply:
edited